File: client/src/app/_models/order.ts
export class Order {
    id: number;
    pizzaType: string;
    pizzaSize: string;
    price: number;
    deliveryAddress: string;
    orderStatus: string;
    phoneNumber: string;
    firstName: string;
    lastName: string;
    estimatedTime: number;
    dateTimeDelivered?: string;
    dateTimeDelivering?: string;
    dateTimeOrdered: string;

    employeeId?: number;
    tip?: number;

    constructor(pizzaType: string, pizzaSize: string, price: number, deliveryAddress: string, orderStatus: string, phoneNumber: string, firstName: string, lastName: string, estimatedTime: number, employeeId?: number, tip?: number) {
        this.pizzaType = pizzaType;
        this.pizzaSize = pizzaSize;
        this.price = price;
        this.deliveryAddress = deliveryAddress;
        this.orderStatus = orderStatus;
        this.phoneNumber = phoneNumber;
        this.firstName = firstName;
        this.lastName = lastName;
        this.estimatedTime = estimatedTime;
        this.employeeId = employeeId;
        this.tip = tip;
    }
}

File: Pizza.API/Entities/Order.cs
using System.ComponentModel.DataAnnotations.Schema;

namespace Pizza.API.Entities
{
    public class Order
    {
        public int Id { get; set; }

        public string PizzaType { get; set; }

        public string PizzaSize { get; set; }

        public decimal Price { get; set; }

        public string DeliveryAddress { get; set; }

        public string OrderStatus { get; set; }

        public decimal? Tip { get; set; }

        public int EstimatedTime { get; set; }

        public DateTime DateTimeOrdered { get; set; }

        public DateTime? DateTimeDelivered{ get; set; }

        public DateTime? DateTimeDelivering { get; set; }

        public Employee? Employee { get; set; }

        [ForeignKey("Employee")]
        public int? EmployeeId { get; set; }

        public Customer Customer { get; set; }

        [ForeignKey("Customer")]
        public string PhoneNumber { get; set; }
    }
}
